### ISSUE TITLE
SciPy, NumPy, Matplotlib, Pandas installation and distutils of Python 3.8 error

### ISSUE THREAD

**Issue by annalisasnow** [scipy/scipy#10766]
I am trying to install pre-release Numpy, as I am using python 3.8 I am also trying python 3.8 as  a System environment, without a virtual one. However, there's probably a problem with the distytils, as guys from Python have kindly explained to me. So after running: 
```
$ pip install git+https://github.com/numpy/numpy.git
```
I get:
```
Collecting git+https://github.com/numpy/numpy.git
  Cloning https://github.com/numpy/numpy.git to /private/var/folders/w9/vqby68j57n98vmlc6hk591vm0000gn/T/pip-req-build-0zpbjpo8
  Running command git clone -q https://github.com/numpy/numpy.git /private/var/folders/w9/vqby68j57n98vmlc6hk591vm0000gn/T/pip-req-build-0zpbjpo8
  Running command git submodule update --init --recursive -q
  Installing build dependencies ... error
```

And here the error starts:

```ERROR: Command errored out with exit status 2:
   command: /Library/Frameworks/Python.framework/Versions/3.8/bin/python3.8 /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pip install --ignore-installed --no-user --prefix /private/var/folders/w9/vqby68j57n98vmlc6hk591vm0000gn/T/pip-build-env-5hxny6jy/overlay --no-warn-script-location --no-binary :none: --only-binary :none: -i https://pypi.org/simple -- setuptools wheel cython
       cwd: None
  Complete output (23 lines):
  Collecting setuptools
    Using cached https://files.pythonhosted.org/packages/b2/86/095d2f7829badc207c893dd4ac767e871f6cd547145df797ea26baea4e2e/setuptools-41.2.0-py2.py3-none-any.whl
  Collecting wheel
    Using cached https://files.pythonhosted.org/packages/00/83/b4a77d044e78ad1a45610eb88f745be2fd2c6d658f9798a15e384b7d57c9/wheel-0.33.6-py2.py3-none-any.whl
  Collecting cython
  Installing collected packages: setuptools, wheel, cython
  ERROR: Exception:
  Traceback (most recent call last):
    File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pip/_internal/cli/base_command.py", line 188, in main
      status = self.run(options, args)
    File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pip/_internal/commands/install.py", line 398, in run
      installed = install_given_reqs(
    File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pip/_internal/req/__init__.py", line 54, in install_given_reqs
      requirement.install(
    File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pip/_internal/req/req_install.py", line 925, in install
      self.move_wheel_files(
    File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pip/_internal/req/req_install.py", line 453, in move_wheel_files
      wheel.move_wheel_files(
    File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pip/_internal/wheel.py", line 324, in move_wheel_files
      scheme = distutils_scheme(
    File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pip/_internal/locations.py", line 104, in distutils_scheme
      assert not (home and prefix), "home={} prefix={}".format(home, prefix)
  AssertionError: home=/private/var/folders/w9/vqby68j57n98vmlc6hk591vm0000gn/T/pip-target-qb0qk201 prefix=/private/var/folders/w9/vqby68j57n98vmlc6hk591vm0000gn/T/pip-build-env-5hxny6jy/overlay
  ----------------------------------------
ERROR: Command errored out with exit status 2: /Library/Frameworks/Python.framework/Versions/3.8/bin/python3.8 /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pip install --ignore-installed --no-user --prefix /private/var/folders/w9/vqby68j57n98vmlc6hk591vm0000gn/T/pip-build-env-5hxny6jy/overlay --no-warn-script-location --no-binary :none: --only-binary :none: -i https://pypi.org/simple -- setuptools wheel cython Check the logs for full command output.
```

I have tried installing
```
setuptools, wheel, cython
``` 
via pip, e.g.: 
```
pip install wheel setuptools cython
``` 
that works perfectly. 

If I run the interim commands (see above): 
```
git clone -q https://github.com/numpy/numpy.git /private/var/folders/w9/vqby68j57n98vmlc6hk591vm0000gn/T/pip-req-build-0zpbjpo8
```
and
 ```
git submodule update --init --recursive -q
``` 
these work too and finally I get the 
```
pip-build-env-5hxny6jy
```
 file. However, if I try running: 
```
/Library/Frameworks/Python.framework/Versions/3.8/bin/python3.8 /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pip install --ignore-installed --no-user --prefix /private/var/folders/w9/vqby68j57n98vmlc6hk591vm0000gn/T/pip-build-env-oepwn_cy/overlay --no-warn-script-location --no-binary :none: --only-binary :none: -i https://pypi.org/simple -- setuptools wheel cython
```
 I am getting errors back.

I was able to install Numpy, SciPy, Matplotlib and Pandas manually, like that:
```
pip download <somewhat>
cd somewhat
python setup.py build
python setup.py install
```
That way I was able to install actually current versions to Python3.8!

However, I see problems with the pre-release installation. What can be done in that case?
#### Scipy/Numpy/Python version information:
```
import sys, scipy, numpy; print(scipy.__version__, numpy.__version__, sys.version_info)
>1.4.0.dev0+ 1.18.0.dev0+6177cbe sys.version_info(major=3, minor=8, micro=0, releaselevel='beta', serial=4)
Python version: 3.8.0b4
Pip version: pip 19.2.3 from /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pip (python 3.8)
```

**Comment by andyfaff** [comment]
This seems to be an issue for pre-release installation of numpy. It would be better to direct this question at numpy rather than scipy. 
From a cursory glance it seems like there's an issue installing cython. Can you build from the numpy source rather than pip installing it?

**Comment by annalisasnow** [comment]
I have already loaded the libs with pip and installed them. But just for the sake of an experiment I have loaded sources of Numpy 1.16, 1.17 and 1.18.0.dev0+6bdbc1e I can build all of them from the source, though I think, the issue with cython and other libs still exist. It is also worth mentioning, I can install cython like that: 
```
pip install cython
```
without any issues at all.
Also, that issue appears when I try installing SciPy the same way as NumPy:
```pip install git+https://github.com/scipy/scipy.git
Collecting git+https://github.com/scipy/scipy.git
  Cloning https://github.com/scipy/scipy.git to /private/var/folders/w9/vqby68j57n98vmlc6hk591vm0000gn/T/pip-req-build-xwrhk381
  Running command git clone -q https://github.com/scipy/scipy.git /private/var/folders/w9/vqby68j57n98vmlc6hk591vm0000gn/T/pip-req-build-xwrhk381
  Running command git submodule update --init --recursive -q
  Installing build dependencies ... error
```
And here comes the error log:
```
  ERROR: Command errored out with exit status 2:
   command: /Library/Frameworks/Python.framework/Versions/3.8/bin/python3.8 /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pip install --ignore-installed --no-user --prefix /private/var/folders/w9/vqby68j57n98vmlc6hk591vm0000gn/T/pip-build-env-6k1352fo/overlay --no-warn-script-location --no-binary :none: --only-binary :none: -i https://pypi.org/simple -- wheel setuptools 'Cython>=0.29.2' 'numpy==1.13.3; python_version=='"'"'3.5'"'"' and platform_system!='"'"'AIX'"'"'' 'numpy==1.13.3; python_version=='"'"'3.6'"'"' and platform_system!='"'"'AIX'"'"'' 'numpy==1.14.5; python_version>='"'"'3.7'"'"' and platform_system!='"'"'AIX'"'"'' 'numpy==1.16.0; python_version=='"'"'3.5'"'"' and platform_system=='"'"'AIX'"'"'' 'numpy==1.16.0; python_version=='"'"'3.6'"'"' and platform_system=='"'"'AIX'"'"'' 'numpy==1.16.0; python_version>='"'"'3.7'"'"' and platform_system=='"'"'AIX'"'"'' 'pybind11>=2.2.4'
       cwd: None
  Complete output (31 lines):
  Ignoring numpy: markers 'python_version == "3.5" and platform_system != "AIX"' don't match your environment
  Ignoring numpy: markers 'python_version == "3.6" and platform_system != "AIX"' don't match your environment
  Ignoring numpy: markers 'python_version == "3.5" and platform_system == "AIX"' don't match your environment
  Ignoring numpy: markers 'python_version == "3.6" and platform_system == "AIX"' don't match your environment
  Ignoring numpy: markers 'python_version >= "3.7" and platform_system == "AIX"' don't match your environment
  Collecting wheel
    Using cached https://files.pythonhosted.org/packages/00/83/b4a77d044e78ad1a45610eb88f745be2fd2c6d658f9798a15e384b7d57c9/wheel-0.33.6-py2.py3-none-any.whl
  Collecting setuptools
    Using cached https://files.pythonhosted.org/packages/b2/86/095d2f7829badc207c893dd4ac767e871f6cd547145df797ea26baea4e2e/setuptools-41.2.0-py2.py3-none-any.whl
  Collecting Cython>=0.29.2
  Collecting numpy==1.14.5
  Collecting pybind11>=2.2.4
    Downloading https://files.pythonhosted.org/packages/5d/85/c7a8dffda52ce25a8bcfe9a28b6861bdd52da59ae001fdd4173e054b7d9b/pybind11-2.3.0-py2.py3-none-any.whl (147kB)
  Installing collected packages: wheel, setuptools, Cython, numpy, pybind11
  ERROR: Exception:
  Traceback (most recent call last):
    File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pip/_internal/cli/base_command.py", line 188, in main
      status = self.run(options, args)
    File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pip/_internal/commands/install.py", line 398, in run
      installed = install_given_reqs(
    File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pip/_internal/req/__init__.py", line 54, in install_given_reqs
      requirement.install(
    File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pip/_internal/req/req_install.py", line 925, in install
      self.move_wheel_files(
    File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pip/_internal/req/req_install.py", line 453, in move_wheel_files
      wheel.move_wheel_files(
    File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pip/_internal/wheel.py", line 324, in move_wheel_files
      scheme = distutils_scheme(
    File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pip/_internal/locations.py", line 104, in distutils_scheme
      assert not (home and prefix), "home={} prefix={}".format(home, prefix)
  AssertionError: home=/private/var/folders/w9/vqby68j57n98vmlc6hk591vm0000gn/T/pip-target-x8c5t_4l prefix=/private/var/folders/w9/vqby68j57n98vmlc6hk591vm0000gn/T/pip-build-env-6k1352fo/overlay
  ----------------------------------------
ERROR: Command errored out with exit status 2: /Library/Frameworks/Python.framework/Versions/3.8/bin/python3.8 /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pip install --ignore-installed --no-user --prefix /private/var/folders/w9/vqby68j57n98vmlc6hk591vm0000gn/T/pip-build-env-6k1352fo/overlay --no-warn-script-location --no-binary :none: --only-binary :none: -i https://pypi.org/simple -- wheel setuptools 'Cython>=0.29.2' 'numpy==1.13.3; python_version=='"'"'3.5'"'"' and platform_system!='"'"'AIX'"'"'' 'numpy==1.13.3; python_version=='"'"'3.6'"'"' and platform_system!='"'"'AIX'"'"'' 'numpy==1.14.5; python_version>='"'"'3.7'"'"' and platform_system!='"'"'AIX'"'"'' 'numpy==1.16.0; python_version=='"'"'3.5'"'"' and platform_system=='"'"'AIX'"'"'' 'numpy==1.16.0; python_version=='"'"'3.6'"'"' and platform_system=='"'"'AIX'"'"'' 'numpy==1.16.0; python_version>='"'"'3.7'"'"' and platform_system=='"'"'AIX'"'"'' 'pybind11>=2.2.4' Check the logs for full command output.
```
Should I still move that to NumPy or probably ask people from Cython?

**Comment by rgommers** [comment]
I don't understand the exception, but it does seem like this simply won't work at the moment. numpy 1.14.5 is not compatible with Python 3.8 AFAIK. Once we have a NumPy release that's Python 3.8 compatible, we will update SciPy's `pyproject.toml` with that version, and then things should work. For now, you should just use `python setup.py install` only with Python 3.8

tl;dr there's no bug, but don't expect `pip` to work for a Python version that's not yet released.
